crates/solver/src/driver.rs

```diff
@@ -179,10 +180,11 @@ impl Driver {
                 // Since we simulate and only submit solutions when they used to pass before, there is no
                 // point in logging transaction failures in the form of race conditions as hard errors.
                 let name = solver.name();
-                if err
-                    .downcast_ref::<MethodError>()
-                    .map(|e| is_transaction_failure(&e.inner))
-                    .unwrap_or(false)
+                if err.is::<TransactionTimeoutError>()
```

Comment:
I'm a bit worried we are splitting the logic which tokens should not be errored upon and the fact that those are reported otherwise in two very different places.

Would it be a lot of work to refactor this error into a custom type that can have a `log_severity` or `should_error` flag?

Alternatively we could the logging of such errors here instead of in the submission module below.